Source/WebCore/css/StyleProperties.cpp

```diff
@@ -226,11 +235,8 @@ String StyleProperties::getPropertyValue(CSSPropertyID propertyID) const
         }
     }
 
-    {
-        auto shorthandValue = String();
-        if (shorthandHasVariableReference(propertyID, shorthandValue))
-            return shorthandValue;
-    }
+    if (std::optional<String> shorthandValue = commonShorthandChecks(propertyID))
```

Comment:
Since `String` has a distinct null and empty value I suggest not bothering with `optional`. The function should return `emptyString()` for cases that should serialize as the empty string and null string if there is no special consideration. The code can be:
```
if (auto result = commonShorthandChecks(propertyID); !result.isNull())
    return result;
```

Reply:
But this seems more risky, e.g. in `StyleProperties::asTextInternal()` it breaks

```cpp
            value = getPropertyValue(shorthandPropertyID);
            if (!value.isNull()) {
```

There are lots of callers, most of which return the value to even more callers, so it seems easy to miss something. This seems safer:

```cpp
if (auto result = commonShorthandChecks(propertyID); !result.isNull())
    return result.isEmpty() ? nullString() : result;
```

Reply:
Fine for your change now, but not a great situation.

Now that you pointed this out I think the code is fragile right now. Functions like `fontValue` are returning empty strings to mean nit serializable, not null string. We should probably change `asTextInternal` to check for empty, not null, or find all the places using empty string and change them to consistently use null. Maybe even find a place to assert that.

The primary use of `getPropertyValue`, where it’s exposed to the bindings, will treat empty string and null string the same. If we want to use null string to mean something special internally, we should figure out what that is. Maybe null should mean “property literally does not exist” in some stronger sense than “serializes as null’.

Reply:
Luckily `asTextInternal()` doesn't attempt to use `CSSPropertyFont`, but it seems fragile indeed.